### PR TITLE
[Performance] Add cached resource metadata collection factory

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

Capture d’écran du 2023-03-24 10-26-24

### CODE REVIEW

File: src/Bundle/Resources/config/services/metadata.xml [hunk 1]
```diff
@@ -13,16 +13,30 @@
 
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
+        <service id="sylius.cache.metadata.resource_collection" parent="cache.system" public="false">
```
Comment (loic425) [inline src/Bundle/Resources/config/services/metadata.xml:16]: @lchrusciel 
the services in this file are private by default, but I think it works differently with parent, cause this one is public if I do not tag it as private.

**Comment by loic425** [comment]
@lchrusciel After this PR, it would be interesting to do some Blackfire sessions :dancers:

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
But we should do it before and after 😂

**Comment by lchrusciel** [comment]
Thank you, @loic425!

**Comment by loic425** [comment]
> But we should do it before and after 😂

Yes but I'd like to compare resource controller vs operations performances too.

**Pull request merged**